lib/encoder_impl.cc

```diff
@@ -286,6 +287,12 @@ void encoder_impl::set_ps(std::string ps) {
 
 		std::memset(PS, ' ', sizeof(PS));
 		std::memcpy(PS, ps.c_str(), len);
+
+        using std::cout;
+        using std::endl;
+        cout << "PS set to \"" << ps << "\"" << endl;
+        rebuild();
+        rebuild();
```

Comment:
I see ndenting with tabs as bad practice since it's not producing same results when working in teams. I didn't find a codingstyle guideline. Feel free to run this through indent and change it to your favourite style.